### ISSUE TITLE
Conflict in data type for custom code using Library

### ISSUE THREAD

**Issue by syg-eliseu** [FlutterFlow/flutterflow-issues#4998]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

The error occurs if I use a Google Map in the project and pass a function from the library that returns a LatLng as a parameter. Since the map definition in the project uses LatLng, when I try to use something in the project that returns this type of data, it interprets it as a different type, causing a type conflict. Log below:

> lib/pages/home_page/home_page_widget.dart:140:69: Error: A value of type 'LatLng/*1*/' can't be assigned to a variable of type 'LatLng/*2*/?'.
> - 'LatLng/*1*/' is from 'package:test_library_ydxklu/flutter_flow/lat_lng.dart' ('dependencies/test_library_ydxklu/lib/flutter_flow/lat_lng.dart').
> - 'LatLng/*2*/' is from 'package:test/flutter_flow/lat_lng.dart' ('lib/flutter_flow/lat_lng.dart').
                      test_library_ydxklu_functions.testeLatLngLib()!,

### Expected Behavior

I hope FlutterFlow can handle the data types received throughout the project so that I can use the custom code from my library as I previously did with Team Code. In its current state, using Libraries is unfeasible for us.

### Steps to Reproduce

1. Create a new FF project. 
2. Add a function to the project that returns a `LatLng`, and inside the function, include only `return new LatLng(0, 0);` for testing purposes.  
3. In a project screen, add a Google Map and use the function as the map's parameter.  
4. Duplicate the project, rename the created function in the duplicate, and then publish it as a library.  
5. Open the initial project and import the library.  
6. In the project, add a new map that uses the function from the library this time.  
7. Enable the web platform in the project.  
8. Run the local execution with Chrome.  
9. You will see the error in the Device Logs panel.  

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITESz/Hqx89gochK+s/1bvpqhQMnQzM5U+cFjtt+Zw8hfIznE5oLeM/CbhdLYtfnSn56OlaknnwKp8LTjvL9A8coBxaqbIBg1a5XQQHJfF+VbKa6PbiWSXElJJ9jfFSm4Z2jmhJRNfVqSXwl23GufvCrG3qCf9qOYwx5e6fDbOY=

### Visual documentation

![image](https://github.com/user-attachments/assets/9e640d29-f88b-4d37-81b1-434cdd670a5a)

### Environment

```markdown
- FlutterFlow version: 5.0.12+
- Platform: Windows Desktop
- Browser name and version:
- Operating system and version affected: Windows 11
```

### Additional Information

I identified the error occurring with the types:  
- `LatLng`  
- `UploadedFile`  

But it may also be happening with other types.

**Comment by Alezanello** [comment]
Thank you so much for bringing this to our attention! I’ll make sure to forward it to our engineering team so they can work on a fix. We really appreciate your feedback!

**Comment by syg-eliseu** [comment]
Hello, when is this fix expected to be available?

**Comment by Alezanello** [comment]
The fix is already finished we are waiting for a production release, it supposed to be yesterday but we got a little delay and could it be done on Thursday!<br><br>Happy Holidays!

**Comment by syg-eliseu** [comment]
Thank you, I look forward to it. Happy holidays to you too.

**Comment by bbeckford** [comment]
Is this update out now? My project still won't compile with the above error.

**Comment by syg-eliseu** [comment]
I can't compile either. I received several updates in the last week, but none of them brought improvements.

![image](https://github.com/user-attachments/assets/3548ec06-a29f-44c7-8ce6-933b04b73274)

**Comment by syg-eliseu** [comment]
Unfortunately, the update you made did not solve the problem, it generated another one.

With this update, the project's custom function generates an error that did not exist before, as if the LatLng class did not exist.

This only happens when I import the library. If I don't import a library, the function doesn't throw errors. If I import an empty library, without LatLng type functions, it still generates errors in my project's LatLng functions.

In other words, we are unable to use libraries with projects that contain custom functions that return LatLng and File values.

Please fix this.

Note: Translated from Portuguese.

**With imported library**
![image](https://github.com/user-attachments/assets/b165e248-d41d-414a-b50e-1d2796d622da)

**No imported library**
![image](https://github.com/user-attachments/assets/964308d1-0366-4716-904b-9d90ec00fe8c)

**Comment by bbeckford** [comment]
After upgrading my project I'm also now getting the same secondary issue as @syg-eliseu 

![image](https://github.com/user-attachments/assets/df334507-46ad-49d2-88c2-2d75f08ba431)
